test/verify/check-client

```diff
@@ -0,0 +1,211 @@
+        # replicate the plumbing bits of src/client/cockpit-client to set up cockpit-client-ssh
+        m.write("/etc/cockpit/cockpit.conf", """
+[WebService]
+X-For-CockpitClient = true
+LoginTo = true
+
+[Ssh-Login]
+ReportStderr = true
+Command = /usr/local/bin/cockpit-client-ssh
+""", append=True)
+        # fake flatpak-spawn wrapper
+        m.write("/usr/local/bin/flatpak-spawn", """#!/bin/sh -eu
+[ "$1" = "--host" ] || exit 1
+shift
+exec "$@"
+""", perm="755")
+
+        m.stop_cockpit()
+        m.spawn(f"runuser -u admin -- {self.libexecdir}/cockpit-ws --no-tls", "ws.log")
+        # FIXME: move to ~/.cache, and create directory in c-client-ssh
+        m.execute("runuser -u admin -- mkdir -p ~admin/.local/share")
+
+        # set up target machine
+        self.machines["target"].execute("hostnamectl set-hostname target")
+
+        self.checkLoginScenarios()
+
+    def testBeiboot(self):
+        m = self.machines["client"]
+
+        m.upload(["../src/client/cockpit-beiboot"], "/usr/local/bin/")
+        # replicate the plumbing bits of src/client/cockpit-client to set up cockpit-beiboot
+        m.write("/etc/cockpit/cockpit.conf", """
+[WebService]
+X-For-CockpitClient = true
+LoginTo = true
+
+[Ssh-Login]
+ReportStderr = true
+Command = /usr/local/bin/cockpit-beiboot
+""", append=True)
+        m.stop_cockpit()
+        m.spawn(f"runuser -u admin -- {self.libexecdir}/cockpit-ws --no-tls", "ws.log")
+        # FIXME: create cache dir in the client
+        m.execute("runuser -u admin -- mkdir -p ~admin/.cache")
+
+        # set up target machine: no cockpit
+        mt = self.machines["target"]
+        mt.execute("rm /usr/bin/cockpit-bridge; rm -r /usr/share/cockpit")
+        mt.execute("hostnamectl set-hostname target")
+
+        self.checkLoginScenarios()
+
+    def checkLoginScenarios(self):
+        m = self.machine
+        mt = self.machines["target"]
+        b = self.browser
+        b.open("/")
+
+        # same username + password login, unknown host key
+        b.wait_text("#brand", "Connect to:")
+        b.wait_not_visible("#recent-hosts-list")
+        b.set_val("#server-field", "10.111.113.2")
+        b.click("#login-button")
+        b.wait_in_text("#conversation-group", "authenticity of host '10.111.113.2")
+        b.set_val("#conversation-input", "yes")
+        b.click("#login-button")
+        b.wait_text("#conversation-prompt", "admin@10.111.113.2's password: ")
+        b.set_val("#conversation-input", "foobar")
+        b.click("#login-button")
+
+        b.wait_visible('#content')
+        b.wait_in_text("#host-apps", "Services")
+        b.wait_in_text("#host-apps", "Terminal")
+        b.wait_in_text("#host-toggle", "admin@target")
+        self.logout(check_last_host="10.111.113.2")
+
+        # remembers the last host it connected to
+        b.wait_in_text("#recent-hosts-list", "10.111.113.2")
+
+        # same username + password login, now host is known
```

Comment:
I think (based on the code block this one) this should be testing the case where we've remembered in local storage that we have the root credential, but it seems incidental.  Could we get a more explicit testing of this?  This was what caused our main showstopper in the `.0` beta release.